src/GovUk.Education.ExploreEducationStatistics.Public.Data.Processor/Functions/ProcessInitialDataSetVersionFunction.cs

```diff
@@ -43,7 +54,59 @@ public class ProcessInitialDataSetVersionFunction
                 context.InstanceId,
                 input.DataSetVersionId);
 
-            await context.CallActivityAsync(nameof(HandleProcessingFailureFunction.HandleProcessingFailure), input);
+            await context.CallActivityAsync(nameof(HandleProcessingFailure), input.DataSetVersionId);
         }
     }
+
+    [Function(nameof(CompleteProcessing))]
+    public async Task CompleteProcessing(
+        [ActivityTrigger] Guid dataSetVersionId,
+        Guid instanceId,
+        CancellationToken cancellationToken)
+    {
+        var dataSetVersion = await GetDataSetVersion(
+            dataSetVersionId: dataSetVersionId,
+            instanceId: instanceId,
+            cancellationToken);
+
+        var dataSetVersionImport = dataSetVersion.Imports.Single(i => i.InstanceId == instanceId);
+
+        dataSetVersionImport.Stage = DataSetVersionImportStage.Completing;
+        await publicDataDbContext.SaveChangesAsync(cancellationToken);
+
+        // Any additional logic to tidy up after importing will be added here
+
+        dataSetVersion.Status = DataSetVersionStatus.Draft;
+        dataSetVersionImport.Completed = DateTimeOffset.UtcNow;
+        await publicDataDbContext.SaveChangesAsync(cancellationToken);
+    }
+
+    [Function(nameof(HandleProcessingFailure))]
```

Comment:
Is the plan that we'll supply some details as to the error encountered here?

Reply:
I think we could do this and stash some error details in the `DataSetVersionImport`, but we should also be able to get any exception details from the durable orchestrations build in HTTP API's to retrieve the status of the instance. That's the reason why I've stashed the instance id on the `DataSetVersionImport`.

I've spoken with @ntsim about this previously and I think we said there could be a proxy in the Admin to that built-in API to retrieve the status from the Processor function app, abstracting away some of the complexity of it.

For now though, we haven't done anything like that so we would need to query the built-in API directly with the instance id to get the error.

Btw, EES-5096 hasn't implemented the `HandleProcessingFailure` function, it's only moved the location of it, which is why your seeing this change here.

Reply:
Yup no probs, I figured, thanks!